### ISSUE TITLE
Filter is missing from schemas output from exportGqlSchemaPath

### ISSUE THREAD

**Issue by james-ff** [graphile/postgraphile#828]
I'm submitting a ... 
* [x] bug report (possibly)

PostGraphile version: 
- 4.0.0-rc.3

postgraphile-plugin-connection-filter version: 
- 1.0.0-beta.14

Steps to reproduce:

I'm using a React frontend with Apollo Client as a GraphQL client that connects to a postgraphile api. For some reason "filter" arguments give the error below, but other filter type arguments like "condition" or "after" do not. This did not happen in the past, so I'm now seeing errors on previously working queries. Oddly this is happening in Apollo Client but not GraphiQL (for the same query), I can't tell if this is due to Apollo Client or some config in the Postgraphile schema.

Error:
`
[GraphQL error]: Message: Unknown argument "filter" on field "allTransfers" of type "Query". Did you mean "after"?, Location: [object Object], Path: undefined
`

Current behavior:
- Passing "filter" as an argument to appropriate fields gives an error on Apollo Client.

Expected behavior:
- No Errors

**Comment by sjmcdowall** [comment]
How about including the whole graphql query ?? 

And if it did not happen in the past -- what versions of the libraries changed?  Besides postgraphile -- how about the React version,  Apollo Client version?  Did the underlying schema change?  What all has changed? 

Interesting that GraphiQL accepts and returns the data correctly..

**Comment by benjie** [comment]
The error you have there is from GraphQL; it's basically saying that "filter" is not part of your GraphQL schema in that location. I think your Apollo client is talking to a different GraphQL endpoint than GraphiQL is - probably one that doesn't have the filter plugin loaded. This implies you're running two versions of PostGraphile somehow?

**Comment by james-ff** [comment]
After more exploring I have the following conclusions
- This is not Apollo Client
- This is likely an issue with the schema file output by postgraphile

For my use case I use the output of `exportGqlSchemaPath` to create a remote executable schema for a postgraphile instance (so that I can merge the schema with other schemas / write my own backend logic / connect to other graphql endpoints).

I have searched the output of this file and I can't find "filter" anywhere in the text of the file, I do however see all of the other filter-like arguments. This is the reason filter works with the graphiql directly linked to the postgraphile instance, but does not work in the playground linked to my final API.

Could you guys check why filter is not being included in the generated schema output?

![screen shot 2018-08-10 at 19 09 32](https://user-images.githubusercontent.com/1874352/43973960-efebe2da-9cd0-11e8-95ce-1e87eaed171b.png)

![screen shot 2018-08-10 at 19 06 39](https://user-images.githubusercontent.com/1874352/43973904-b99e6248-9cd0-11e8-881d-ffe0d716f84b.png)

![screen shot 2018-08-10 at 18 51 48](https://user-images.githubusercontent.com/1874352/43973147-91a4485e-9cce-11e8-8a29-b9867c4746c6.png)

**Comment by james-ff** [comment]
TL;DR

`exportGqlSchemaPath` is not including 'filter' attributes in the schema output

**Comment by benjie** [comment]
I'm sorry; I cannot reproduce this issue. Please provide a reproducible example.

Here's what I did:

```
mkdir test828
cd test828
npm install postgraphile postgraphile-plugin-connection-filter express
createdb issue828
echo "create table foo (id serial primary key);" | psql issue828
```

Then I put this in server.js:

```js
const express = require("express");
const { postgraphile } = require("postgraphile");
const PostGraphileConnectionFilterPlugin = require("postgraphile-plugin-connection-filter");


const app = express();

app.use(postgraphile("postgres://localhost/issue828", ["public"], {
  appendPlugins: [
    PostGraphileConnectionFilterPlugin
  ],
  exportGqlSchemaPath: `${__dirname}/schema.graphql`,
}));

app.listen(process.env.PORT || 3000);
```

Then I ran server.js via `node server.js` and that generated the schema.graphql file that contained:

```graphql
type Query implements Node {
#...
  """Reads and enables pagination through a set of `Foo`."""
  allFoos(
    """Only read the first `n` values of the set."""
    first: Int

    """Only read the last `n` values of the set."""
    last: Int

    """
    Skip the first `n` values from our `after` cursor, an alternative to cursor
    based pagination. May not be used with `last`.
    """
    offset: Int

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """The method to use when ordering `Foo`."""
    orderBy: [FoosOrderBy!] = [PRIMARY_KEY_ASC]

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: FooCondition

    """
    A filter to be used in determining which values should be returned by the collection.
    """
    filter: FooFilter
  ): FoosConnection
#...
}
```

As you can see, the `filter` option is available on the allFoos connection, as expected.


<details>
<summary>Full schema</summary>

```graphql
"""All input for the create `Foo` mutation."""
input CreateFooInput {
  """
  An arbitrary string value with no semantic meaning. Will be included in the
  payload verbatim. May be used to track mutations by the client.
  """
  clientMutationId: String

  """The `Foo` to be created by this mutation."""
  foo: FooInput!
}

"""The output of our create `Foo` mutation."""
type CreateFooPayload {
  """
  The exact same `clientMutationId` that was provided in the mutation input,
  unchanged and unused. May be used by a client to track mutations.
  """
  clientMutationId: String

  """The `Foo` that was created by this mutation."""
  foo: Foo

  """
  Our root query field type. Allows us to run any query from our mutation payload.
  """
  query: Query

  """An edge for our `Foo`. May be used by Relay 1."""
  fooEdge(
    """The method to use when ordering `Foo`."""
    orderBy: [FoosOrderBy!] = PRIMARY_KEY_ASC
  ): FoosEdge
}

"""A location in a connection that can be used for resuming pagination."""
scalar Cursor

"""All input for the `deleteFooById` mutation."""
input DeleteFooByIdInput {
  """
  An arbitrary string value with no semantic meaning. Will be included in the
  payload verbatim. May be used to track mutations by the client.
  """
  clientMutationId: String
  id: Int!
}

"""All input for the `deleteFoo` mutation."""
input DeleteFooInput {
  """
  An arbitrary string value with no semantic meaning. Will be included in the
  payload verbatim. May be used to track mutations by the client.
  """
  clientMutationId: String

  """
  The globally unique `ID` which will identify a single `Foo` to be deleted.
  """
  nodeId: ID!
}

"""The output of our delete `Foo` mutation."""
type DeleteFooPayload {
  """
  The exact same `clientMutationId` that was provided in the mutation input,
  unchanged and unused. May be used by a client to track mutations.
  """
  clientMutationId: String

  """The `Foo` that was deleted by this mutation."""
  foo: Foo
  deletedFooId: ID

  """
  Our root query field type. Allows us to run any query from our mutation payload.
  """
  query: Query

  """An edge for our `Foo`. May be used by Relay 1."""
  fooEdge(
    """The method to use when ordering `Foo`."""
    orderBy: [FoosOrderBy!] = PRIMARY_KEY_ASC
  ): FoosEdge
}

type Foo implements Node {
  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!
  id: Int!
}

"""
A condition to be used against `Foo` object types. All fields are tested for equality and combined with a logical ‘and.’
"""
input FooCondition {
  """Checks for equality with the object’s `id` field."""
  id: Int
}

"""
A filter to be used against `Foo` object types. All fields are combined with a logical ‘and.’
"""
input FooFilter {
  """Filter by the object’s `id` field."""
  id: IntFilter

  """Checks for all expressions in this list."""
  and: [FooFilter!]

  """Checks for any expressions in this list."""
  or: [FooFilter!]

  """Negates the expression."""
  not: FooFilter
}

"""An input for mutations affecting `Foo`"""
input FooInput {
  id: Int
}

"""Represents an update to a `Foo`. Fields that are set will be updated."""
input FooPatch {
  id: Int
}

"""A connection to a list of `Foo` values."""
type FoosConnection {
  """A list of `Foo` objects."""
  nodes: [Foo]!

  """
  A list of edges which contains the `Foo` and cursor to aid in pagination.
  """
  edges: [FoosEdge!]!

  """Information to aid in pagination."""
  pageInfo: PageInfo!

  """The count of *all* `Foo` you could get from the connection."""
  totalCount: Int
}

"""A `Foo` edge in the connection."""
type FoosEdge {
  """A cursor for use in pagination."""
  cursor: Cursor

  """The `Foo` at the end of the edge."""
  node: Foo
}

"""Methods to use when ordering `Foo`."""
enum FoosOrderBy {
  NATURAL
  ID_ASC
  ID_DESC
  PRIMARY_KEY_ASC
  PRIMARY_KEY_DESC
}

"""
A filter to be used against Int fields. All fields are combined with a logical ‘and.’
"""
input IntFilter {
  """
  If set to true, checks for null values.  If set to false, checks for non-null values.
  """
  isNull: Boolean

  """Checks for values equal to this value."""
  equalTo: Int

  """Checks for values not equal to this value."""
  notEqualTo: Int

  """
  Checks for values not equal to this value, treating null like an ordinary value.
  """
  distinctFrom: Int

  """
  Checks for values equal to this value, treating null like an ordinary value.
  """
  notDistinctFrom: Int

  """Checks for values less than this value."""
  lessThan: Int

  """Checks for values less than or equal to this value."""
  lessThanOrEqualTo: Int

  """Checks for values greater than this value."""
  greaterThan: Int

  """Checks for values greater than or equal to this value."""
  greaterThanOrEqualTo: Int

  """Checks for values in this list."""
  in: [Int!]

  """Checks for values not in this list."""
  notIn: [Int!]
}

"""
The root mutation type which contains root level fields which mutate data.
"""
type Mutation {
  """Creates a single `Foo`."""
  createFoo(
    """
    The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
    """
    input: CreateFooInput!
  ): CreateFooPayload

  """Updates a single `Foo` using its globally unique id and a patch."""
  updateFoo(
    """
    The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
    """
    input: UpdateFooInput!
  ): UpdateFooPayload

  """Updates a single `Foo` using a unique key and a patch."""
  updateFooById(
    """
    The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
    """
    input: UpdateFooByIdInput!
  ): UpdateFooPayload

  """Deletes a single `Foo` using its globally unique id."""
  deleteFoo(
    """
    The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
    """
    input: DeleteFooInput!
  ): DeleteFooPayload

  """Deletes a single `Foo` using a unique key."""
  deleteFooById(
    """
    The exclusive input argument for this mutation. An object type, make sure to see documentation for this object’s fields.
    """
    input: DeleteFooByIdInput!
  ): DeleteFooPayload
}

"""An object with a globally unique `ID`."""
interface Node {
  """
  A globally unique identifier. Can be used in various places throughout the system to identify this single value.
  """
  nodeId: ID!
}

"""Information about pagination in a connection."""
type PageInfo {
  """When paginating forwards, are there more items?"""
  hasNextPage: Boolean!

  """When paginating backwards, are there more items?"""
  hasPreviousPage: Boolean!

  """When paginating backwards, the cursor to continue."""
  startCursor: Cursor

  """When paginating forwards, the cursor to continue."""
  endCursor: Cursor
}

"""The root query type which gives access points into the data universe."""
type Query implements Node {
  """
  Exposes the root query type nested one level down. This is helpful for Relay 1
  which can only query top level fields if they are in a particular form.
  """
  query: Query!

  """
  The root query type must be a `Node` to work well with Relay 1 mutations. This just resolves to `query`.
  """
  nodeId: ID!

  """Fetches an object given its globally unique `ID`."""
  node(
    """The globally unique `ID`."""
    nodeId: ID!
  ): Node

  """Reads and enables pagination through a set of `Foo`."""
  allFoos(
    """Only read the first `n` values of the set."""
    first: Int

    """Only read the last `n` values of the set."""
    last: Int

    """
    Skip the first `n` values from our `after` cursor, an alternative to cursor
    based pagination. May not be used with `last`.
    """
    offset: Int

    """Read all values in the set before (above) this cursor."""
    before: Cursor

    """Read all values in the set after (below) this cursor."""
    after: Cursor

    """The method to use when ordering `Foo`."""
    orderBy: [FoosOrderBy!] = [PRIMARY_KEY_ASC]

    """
    A condition to be used in determining which values should be returned by the collection.
    """
    condition: FooCondition

    """
    A filter to be used in determining which values should be returned by the collection.
    """
    filter: FooFilter
  ): FoosConnection
  fooById(id: Int!): Foo

  """Reads a single `Foo` using its globally unique `ID`."""
  foo(
    """The globally unique `ID` to be used in selecting a single `Foo`."""
    nodeId: ID!
  ): Foo
}

"""All input for the `updateFooById` mutation."""
input UpdateFooByIdInput {
  """
  An arbitrary string value with no semantic meaning. Will be included in the
  payload verbatim. May be used to track mutations by the client.
  """
  clientMutationId: String

  """
  An object where the defined keys will be set on the `Foo` being updated.
  """
  fooPatch: FooPatch!
  id: Int!
}

"""All input for the `updateFoo` mutation."""
input UpdateFooInput {
  """
  An arbitrary string value with no semantic meaning. Will be included in the
  payload verbatim. May be used to track mutations by the client.
  """
  clientMutationId: String

  """
  The globally unique `ID` which will identify a single `Foo` to be updated.
  """
  nodeId: ID!

  """
  An object where the defined keys will be set on the `Foo` being updated.
  """
  fooPatch: FooPatch!
}

"""The output of our update `Foo` mutation."""
type UpdateFooPayload {
  """
  The exact same `clientMutationId` that was provided in the mutation input,
  unchanged and unused. May be used by a client to track mutations.
  """
  clientMutationId: String

  """The `Foo` that was updated by this mutation."""
  foo: Foo

  """
  Our root query field type. Allows us to run any query from our mutation payload.
  """
  query: Query

  """An edge for our `Foo`. May be used by Relay 1."""
  fooEdge(
    """The method to use when ordering `Foo`."""
    orderBy: [FoosOrderBy!] = PRIMARY_KEY_ASC
  ): FoosEdge
}
```
</details>

**Comment by james-ff** [comment]
Hi @benjie,

I feel really silly, I hadn't appended the filter plugin to the postgraphile options. After adding it the schema output works as intended. Thank you for providing the example, this is what made me realise the cause.

**Comment by benjie** [comment]
Thanks for reporting back 👍
